### PR TITLE
Update soy milk and almond milk to fix carnivores having no problems with them.

### LINKED ISSUE #48843: Carnivores can drink soy and almond milk
Carnivores can currently drink soy and almond milks without issues from them being made from plants.

1. Make/mutate a character with the Carnivore trait.
2. Spawn in some almond or soy milk.
3. Enjoy with whatever meat-based alternatives for cookies.

Carnivores probably shouldn't be able to absorb nutrients from almond or soy milks, because they're plant products.

image

- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 07c58d9 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

# Additional context

### PR DESCRIPTION
**Pull request by RelMayers**
#### Summary
Bugfixes "Update soy milk and almond milk to fix carnivores having no problems with them."

#### Purpose of change
Fixes #48843
Carnivores should not be able to drink what is basically nut juice.

#### Describe the solution
Adding flag nut and veggy to almond and soy milks. (vegetable oil uses veggy flag)

#### Describe alternatives you've considered
Using other flag. (I'm not completly sure there is any interaction caused by this flag)

#### Testing
1.Spawning both soy and almond milk, checked consume menu, changed to carnivore, checked consume menu.
2.Made changes
3.Repeated step 1.

#### Additional context
I'm still new to this so be open with anything you see that i can improve.

### CODE REVIEW

**Review by Maleclypse — APPROVED** [review]
Thanks!

**Comment by actual-nh** [comment]
You can put "Fixes #48843" under "Purpose of change" to automatically close the issue when this is merged, BTW.

**Comment by RelMayers** [comment]
> You can put "Fixes #48843" under "Purpose of change" to automatically close the issue when this is merged, BTW.

thanks

File: src/worldfactory.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include "string_formatter.h"
 #include "string_input_popup.h"
 #include "translations.h"
+#include "try_parse_integer.h"
```
Comment (ZhilkinSerg) [inline src/worldfactory.cpp:34]:
```suggestion
```

**Pull request merged**